hivemind_core/database.py

```
@@ -1,254 +1,688 @@
+            A list of clients that match the search criteria.
+        """
+        # Use index if available
+        if key in ['name', 'api_key']:
+            client_ids = self.redis.smembers(f"client:index:{key}:{val}")
+            res = [cast2client(self.redis.get(f"client:{cid}"))
+                   for cid in client_ids]
+            res = [c for c in res if c.api_key != "revoked"]
+            return res
+
+        res = []
+        for client_id in self.redis.scan_iter(f"client:*"):
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)